### ISSUE TITLE
[Bug]: key.split is not a function

### ISSUE THREAD

**Issue by lost-hope** [bushvin/actor-export#9]
### Contact Details

s.willrodt@yahoo.de

### Foundry VTT version

11.315

### Game system

dnd5e

### Game system version

3.0.4

### actor-export module version

2024.5.3

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Provider is affected?

Dungeons & Dragons 5th edition Character sheets

### What happened?

when trying to export i get a message that "key.split" is not a function. 

### Relevant log output

```Shell
Foundry VTT | Rendering actorExportDialog foundry.js:5802:15
Please wait, this may take a while... foundry.js:68326:78
actor-export | provider: dnd5e main.js:42:25
actor-export | actor: 
Object { name: "Topaz", type: "character", img: "tokenizer/pc-images/topaz.Avatar.webp?1698534590395", system: {…}, prototypeToken: {…}, sort: 6000000, ownership: {…}, flags: {…}, _stats: {…}, overrides: {…}, … }
main.js:42:25
actor-export | character: 
Object { game: {…}, actor: {…}, className: "dnd5eActor" }
DND5eHelper.js:21:17
actor-export | An error ocurred executing the 'dnd5e' provider: key.split is not a function foundry.js:68326:78
Uncaught (in promise) Error: TypeError: key.split is not a function
[Detected 1 package: actor-export]
    downloadFiles http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:471
    promise callback*downloadFiles http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:464
    activateListeners http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:318
    activateListeners http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:316
    _render http://v2202109156138162946.hotsrv.de:30000/scripts/foundry.js:5842
    _render http://v2202109156138162946.hotsrv.de:30000/scripts/foundry.js:6532
    render http://v2202109156138162946.hotsrv.de:30000/scripts/foundry.js:5769
    onclick http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:615
    _renderOuter http://v2202109156138162946.hotsrv.de:30000/scripts/foundry.js:5941
    jQuery 2
main.js:471:31
    downloadFiles http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:471
    (Async: promise callback)
    downloadFiles http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:464
    activateListeners http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:318
    (Async: EventListener.handleEvent)
    activateListeners http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:316
    _render http://v2202109156138162946.hotsrv.de:30000/scripts/foundry.js:5842
    AsyncFunctionNext self-hosted:852
    (Async: async)
    _render http://v2202109156138162946.hotsrv.de:30000/scripts/foundry.js:6532
    render http://v2202109156138162946.hotsrv.de:30000/scripts/foundry.js:5769
    onclick http://v2202109156138162946.hotsrv.de:30000/modules/actor-export/scripts/main.js:615
    _renderOuter http://v2202109156138162946.hotsrv.de:30000/scripts/foundry.js:5941
    jQuery 2
        dispatch
        handle
Foundry VTT | Rendering ModuleManagement foundry.js:5802:15
Foundry VTT | Retrieved and compiled template templates/setup/parts/package-tags.hbs foundry.js:7682:17
Foundry VTT | Retrieved and compiled template templates/sidebar/apps/module-management.html foundry.js:7682:17

​
```

**Comment by bushvin** [comment]
Would it be possible to get an export of your character?

**Comment by lost-hope** [comment]
~~sure if you tell me how to export it correctly~~

 ~~if i try to copy the part from the web dev console, i only get part of it~~
 right click on the actor sheet, if future people stumble over this

**Comment by lost-hope** [comment]
[fvtt-Actor-topaz-Yzg4YmY5Yjk5MWU0.json](https://github.com/bushvin/actor-export/files/15226953/fvtt-Actor-topaz-Yzg4YmY5Yjk5MWU0.json)
nvm found it

**Comment by bushvin** [comment]
Thank you! I am without pc today, so I will check tomorrow.

**Comment by bushvin** [comment]
Hello @lost-hope ,

I believe I have found the issue, and resolved it with the 2024.5.4 release.

If you have any recommendations wrt my module, feel free to let me know!

**Comment by lost-hope** [comment]
Hello,

Yeah it seems fixed now.

Just two other issues: 

The detail sheet shows the GM notes![image-1.png](https://github.com/bushvin/actor-export/assets/6001498/9c7989c5-16b3-41b9-ba6f-0466e4298adc)


And the prepared spells are not correct ![image-2.png](https://github.com/bushvin/actor-export/assets/6001498/986fd35a-696d-4b7c-ad3f-618efcb69f08)

Here are the screenshots of the char sheet:

![image-3.png](https://github.com/bushvin/actor-export/assets/6001498/b0a8fa7a-f5c7-4367-b0c7-b2e922d70f6b)

![image-4.png](https://github.com/bushvin/actor-export/assets/6001498/b694e80c-0cb4-4e1c-aa8e-348d2f601199)

**Comment by bushvin** [comment]
Hey @lost-hope 
Apologies for my tardiness...

**Secret GM info***
I see what you mean with the *secret GM info*. I will need to see what I can do about that, as it is basically markup inside the actor's biography entry.

I imagine this is text you have added as a *section* or so to the biography, and I hope everybody using the DnD5e system does it the same way... Not pointing at you, pointing at the D&D5e System devs. This is IMO bad design.

**Prepared Spells**
Should be an easy fix, I believe all check boxes are one too low.

**Comment by bushvin** [comment]
The issues above should be fixed as well with 2024.5.5

**Comment by bushvin** [comment]
FYI: I also have the german charactersheet now.
